### PR TITLE
mago: init at 0.0.9

### PR DESCRIPTION
**Pull request by gaelreyrol**
This PR initialize [mago](https://github.com/carthage-software/mago) :
> A toolchain for PHP that aims to provide a set of tools to help developers write better code. 

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/ma/mago/package.nix [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  mago,
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "mago";
+  version = "0.0.5";
+
+  src = fetchFromGitHub {
+    owner = "carthage-software";
+    repo = pname;
```
Comment (getchoo) [inline pkgs/by-name/ma/mago/package.nix:15]:
```suggestion
    repo = "mago";
```
https://github.com/NixOS/nixpkgs/issues/277994

File: pkgs/by-name/ma/mago/package.nix [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    rev = version;
```
Comment (getchoo) [inline pkgs/by-name/ma/mago/package.nix:16]:
```suggestion
    tag = version;
```

File: pkgs/by-name/ma/mago/package.nix [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  };
+
+  meta = with lib; {
+    changelog = "https://github.com/carthage-software/mago/releases/tag/${version}";
+    description = "Mago is a toolchain for PHP that aims to provide a set of tools to help developers write better code";
+    homepage = "https://github.com/carthage-software/mago";
+    license = licenses.mit;
+    maintainers = with maintainers; [ gaelreyrol ];
+    mainProgram = "mago";
+  };
```
Comment (getchoo) [inline pkgs/by-name/ma/mago/package.nix:37]:
```suggestion
  meta = {
    changelog = "https://github.com/carthage-software/mago/releases/tag/${version}";
    description = "Mago is a toolchain for PHP that aims to provide a set of tools to help developers write better code";
    homepage = "https://github.com/carthage-software/mago";
    license = lib.licenses.mit;
    maintainers = with lib.maintainers; [ gaelreyrol ];
    mainProgram = "mago";
  };
```
Nit

**Comment by getchoo** [comment]
@ofborg build mago mago.passthru.tests

File: pkgs/by-name/ma/mago/package.nix (hunk 2 above)
Comment (gaelreyrol) [reply pkgs/by-name/ma/mago/package.nix:16]: Are you sure about that ?

File: pkgs/by-name/ma/mago/package.nix (hunk 3 above)
Comment (gaelreyrol) [reply pkgs/by-name/ma/mago/package.nix]: done

File: pkgs/by-name/ma/mago/package.nix (hunk 1 above)
Comment (gaelreyrol) [reply pkgs/by-name/ma/mago/package.nix]: done

File: pkgs/by-name/ma/mago/package.nix (hunk 2 above)
Comment (getchoo) [reply pkgs/by-name/ma/mago/package.nix:16]: Yep, see https://github.com/NixOS/nixpkgs/pull/355973

**Review by getchoo — APPROVED** [review]
LGTM. Feel free to ping in a couple days if there are no other reviews

File: pkgs/by-name/ma/mago/package.nix (hunk 2 above)
Comment (gaelreyrol) [reply pkgs/by-name/ma/mago/package.nix:16]: Thanks for the link :+1:

Comment (gaelreyrol) [reply pkgs/by-name/ma/mago/package.nix]: done

File: pkgs/by-name/ma/mago/package.nix [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    tests.version = testers.testVersion {
+      package = mago;
+      command = "mago --version";
+      version = "mago-cli ${version}";
+    };
+  };
+
+  meta = {
+    changelog = "https://github.com/carthage-software/mago/releases/tag/${version}";
+    description = "Mago is a toolchain for PHP that aims to provide a set of tools to help developers write better code";
```
Comment (Aleksanaa) [inline pkgs/by-name/ma/mago/package.nix:32]:
```suggestion
    description = "Toolchain for PHP that aims to provide a set of tools to help developers write better code";
```

**Review by getchoo — APPROVED** [review]

**Pull request merged**